Address-Book.go

```diff
@@ -0,0 +1,464 @@
+		}
+
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func DeleteNum(context *GlobalData,w http.ResponseWriter, r *http.Request){
+	username :=GetUserFromSession(context,r)
+
+		user := UserContacts{UserName:username}
+
+		err := user.DeleteContactNumber(context.db, r.FormValue("id"), r.FormValue("ID"))
+
+		if err != nil {
+			fmt.Println("DB error")
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (usersession *UserSession) Check( context *GlobalData,w http.ResponseWriter, r *http.Request) {
+
+	user :=GetUserFromSession(context , r)
+	if user !="" {
+		http.Redirect(w,r,"/userpage",http.StatusFound)
+		return
+	}else {
+		http.Redirect(w,r,"/home",http.StatusFound)
+		return
+		}
+	}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func CheckUsernameExists(context *GlobalData , username string) error{
+	var databasePassword string
+
+	err := context.db.Query("SELECT password FROM user_logins WHERE username=?", username).Scan( &databasePassword)
+	fmt.Println(err)
+	return err
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (usersession  *UserSession) HomePage(context *GlobalData, w http.ResponseWriter, r *http.Request) {
+	if err := context.templates.ExecuteTemplate(w, "index.html",usersession.err); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (usersession  *UserSession) SignIn(context *GlobalData, username string, password string , w http.ResponseWriter, r *http.Request ) {
+	// Grab from the database
+	var err error
+	var databasePassword string
+	databasePassword, err = QueryUser(context, username)
+	if err == gocql.ErrNotFound {
+		//no such user
+		usersession.err="Username doesn't exist"
+		http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+		return
+
+	} else if  err != nil {
+		//Database Error
+		usersession.err="Server Error"
+		http.Error(w,  err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = bcrypt.CompareHashAndPassword([]byte(databasePassword), []byte(password))
+	// If wrong password redirect to the login
+	if  err != nil {
+		//Wrong Password
+		usersession.err="Wrong Password"
+		http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+		return
+	} else {
+		// If the login succeeded
+		SaveUserSession(context,username,w,r)
+		http.Redirect(w, r, "/userpage", 301)
+		return
+	}
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (usersession  *UserSession) Register(context *GlobalData,username string ,password string , w http.ResponseWriter, r *http.Request ){
```

Comment:
discuss with mostafa cherri this point you made a user defined type specific to return an error through receiver while you can only make function return parameter of type error instead.
i don't know which is better.